clients/src/test/java/org/apache/kafka/clients/consumer/internals/StreamsGroupHeartbeatRequestManagerTest.java

```diff
@@ -0,0 +1,223 @@
+    private static final String SOURCE_TOPIC_1 = "sourceTopic1";
+    private static final String SOURCE_TOPIC_2 = "sourceTopic2";
+    private static final Set<String> SOURCE_TOPICS = Set.of(SOURCE_TOPIC_1, SOURCE_TOPIC_2);
+    private static final String REPARTITION_SINK_TOPIC_1 = "repartitionSinkTopic1";
+    private static final String REPARTITION_SINK_TOPIC_2 = "repartitionSinkTopic2";
+    private static final String REPARTITION_SINK_TOPIC_3 = "repartitionSinkTopic3";
+    private static final Set<String> REPARTITION_SINK_TOPICS = Set.of(
+        REPARTITION_SINK_TOPIC_1,
+        REPARTITION_SINK_TOPIC_2,
+        REPARTITION_SINK_TOPIC_3
+    );
+    private static final String REPARTITION_SOURCE_TOPIC_1 = "repartitionSourceTopic1";
+    private static final String REPARTITION_SOURCE_TOPIC_2 = "repartitionSourceTopic2";
+    private static final Map<String, StreamsRebalanceData.TopicInfo> REPARTITION_SOURCE_TOPICS = mkMap(
+        mkEntry(REPARTITION_SOURCE_TOPIC_1, new StreamsRebalanceData.TopicInfo(Optional.of(2), Optional.of((short) 1), Map.of("config1", "value1"))),
+        mkEntry(REPARTITION_SOURCE_TOPIC_2, new StreamsRebalanceData.TopicInfo(Optional.of(3), Optional.of((short) 3), Collections.emptyMap()))
+    );
+    private static final String CHANGELOG_TOPIC_1 = "changelogTopic1";
+    private static final String CHANGELOG_TOPIC_2 = "changelogTopic2";
+    private static final String CHANGELOG_TOPIC_3 = "changelogTopic3";
+    private static final Map<String, StreamsRebalanceData.TopicInfo> CHANGELOG_TOPICS = mkMap(
+        mkEntry(CHANGELOG_TOPIC_1, new StreamsRebalanceData.TopicInfo(Optional.empty(), Optional.of((short) 1), Collections.emptyMap())),
+        mkEntry(CHANGELOG_TOPIC_2, new StreamsRebalanceData.TopicInfo(Optional.empty(), Optional.of((short) 2), Collections.emptyMap())),
+        mkEntry(CHANGELOG_TOPIC_3, new StreamsRebalanceData.TopicInfo(Optional.empty(), Optional.of((short) 3), Map.of("config2", "value2")))
+    );
+    private static final Collection<Set<String>> COPARTITION_GROUP = Set.of(
+        Set.of(SOURCE_TOPIC_1, REPARTITION_SOURCE_TOPIC_2),
+        Set.of(SOURCE_TOPIC_2, REPARTITION_SOURCE_TOPIC_1)
+    );
+    private static final String SUBTOPOLOGY_NAME_1 = "subtopology1";
+    private static final StreamsRebalanceData.Subtopology SUBTOPOLOGY_1 = new StreamsRebalanceData.Subtopology(
+        SOURCE_TOPICS,
+        REPARTITION_SINK_TOPICS,
+        REPARTITION_SOURCE_TOPICS,
+        CHANGELOG_TOPICS,
+        COPARTITION_GROUP
+    );
+    private static final Map<String, StreamsRebalanceData.Subtopology> SUBTOPOLOGIES = mkMap(
+        mkEntry(SUBTOPOLOGY_NAME_1, SUBTOPOLOGY_1)
+    );
+    private static final Map<String, String> CLIENT_TAGS = Map.of("clientTag1", "value2");
+
+    private final StreamsRebalanceData streamsRebalanceData = new StreamsRebalanceData(
```

Comment:
We should test also the dynamically updated fields of rebalance data in conjunction with the heartbeat. In particular, setting the reconciled tasks and setting the application shutdown flag. Putting it all into `testSendingFullHeartbeatRequest` would be fine for me though.

Reply:
That would be part of the next PR.